lib/app/features/feed/create_post/views/pages/create_post_modal/components/attached_media_preview_list.dart

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/components/screen_offset/screen_side_offset.dart';
+import 'package:ion/app/extensions/extensions.dart';
+import 'package:ion/app/features/gallery/providers/gallery_provider.c.dart';
+import 'package:ion/app/services/media_service/aspect_ratio.dart';
+import 'package:ion/app/services/media_service/media_service.c.dart';
+import 'package:ion/generated/assets.gen.dart';
+import 'package:photo_manager_image_provider/photo_manager_image_provider.dart';
+
+class AttachedMediaPreview extends ConsumerWidget {
+  const AttachedMediaPreview({
+    required this.attachedMediaNotifier,
+    super.key,
+  });
+
+  final ValueNotifier<List<MediaFile>> attachedMediaNotifier;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final list = attachedMediaNotifier.value;
+
+    final maxItemHeight = 190.0.s;
+    final attachedMediaAspectRatio = list.isEmpty
```

Comment:
I suppose the lines 26-31 can be moved outside of UI

Reply:
To be honest, I'm not sure. `attachedMediaAspectRatio`, `isHorizontalPreviews` directly related to the current widget's UI and will not be reused anywhere else. Can you suggest the better place for it?

Reply:
Added provider for that and class wrapper to handle `isHorizontal`, `isVertical` stuff.